### PR TITLE
refactor(ci): publish packages for PR

### PR DESCRIPTION
**Pull request by GImbrailo**
## 🪂 Pre-Merge Checklist (Definition of Done)

References #1297 
References #1333 

### ✅ Required for Merging

- [ ] 🏄‍♂️ Myself and others have built this branch and done manual testing on the change

### 🙅 Not-Required

- [ ] 🛑 (drag items from above to this list to remove from requirements)
- [ ] 🏖️ New and existing unit tests pass locally and in CI
- [ ] 🔱 The test plan has been implemented and verified by an SDET
- [ ] 🐬 I have commented my code, particularly in hard-to-understand areas
- [ ] 🦀 Automated tests have been written and added to this PR
- [ ] ⛵ Documentation is added/updated for the release version 
- [ ] 🤿 Corresponding changes have been made to all documentation
- [ ] 🖥 Required changes to the VScE have been made
- [ ] 👷 Required updates to scaffolds have been made

----------------------------------------------------------------------------------------------------------------------------

## 🪁 Description

Previously we were publishing npm packages for PRs but this was not done correctly and was later removed. This change adds back this functionality but with the workflow improvements, introduction of `lerna` and `nx`, and our own private Verdaccio registry, we are now able to have working PR packages for testing.

Using `lerna` and `nx` in our workflows make it much easier to uniformly update the version of our packages and their dependencies. It also speeds up our workflows slightly and will make publishing for pre-releases and full-releases a much smoother process. The only step that we will not use `lerns` or `nx` for is the publish step as these tools require setup for `git` and changes to be committed before publishing.

- [x] Setup Verdaccio server
- [x] Package version updated for PRs and publish to Verdaccio registry https://npm.ecadinfra.com
- [x] Make sure that instructions on how to use the packages is in a comment on the PR

## 🎢 Test Plan

More information and steps can be seen in the workflow runs for this PR.

**Published packages on Verdaccio**
image

----------------------------------------------------------------------------------------------------------------------------

### 🛸 Type of Change

- [ ] 🧹 Chore ➾
- [ ] 🛠️ Fix ➾
- [ ] ✨ Feature ➾
- [X] 🏗️ Refactor ➾
- [ ] 🚀 Release ➾

----------------------------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Review by AlirezaHaghshenas — APPROVED** [review]

**Review by AlirezaHaghshenas — APPROVED** [review]

**Review by hu3man — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -8,15 +8,17 @@
 	},
 	"scripts": {
 		"test": "yarn",
-		"build": "./bin/build.sh || npx play-sound beep.mp3",
+		"bootstrap": "npx lerna bootstrap",
```
Comment (danielelisi) [inline package.json:11]: no need to use `npx` when a module is called from an NPM script. Also I would suggest adding `lerna bootstrap --force-local` as it ensures local packages are correctly linked

**Review by danielelisi — APPROVED** [review]
Other then one minor comment, LGTM

File: package.json (hunk 1 above)
Comment (GImbrailo) [reply package.json]: Done, thanks @danielelisi

**Comment by hu3man** [comment]
I'm good with merging this as soon as you are ready @GImbrailo!

**Pull request merged**